pkg/pod/list.go

```diff
@@ -182,3 +207,85 @@ func WaitForAllPodsInNamespaceRunning(
 
 	return true, nil
 }
+
+// WaitForAllPodsInNamespacesHealthy waits until all pods in a list of namespaces that match options
+// are in healthy state.
+// An pod in a healthy state is in running phase and optionally in ready condition.
+//
+// nsNames lists the list of namespaces to monitor. Monitors all namespaces when empty.
+// timeout is the duration to wait for the pods to be healthy
+// includeSucceeded when true, implies that pods in succeeded phase are running.
+// checkReadiness when true, to also checks that the podConditions are ready.
+// ignoreFailedPods when true, to Ignore failed pods with restart policy set to never.
+// ignoreNamespaces is a list of namespaces to ignore.
+// options reduces the list of namespace to only the ones matching options.
+func WaitForAllPodsInNamespacesHealthy(
+	apiClient *clients.Settings,
+	nsNames []string,
+	timeout time.Duration,
+	includeSucceeded bool,
+	skipReadinessCheck bool,
+	ignoreRestartPolicyNever bool,
+	ignoreNamespaces []string,
+	options ...metav1.ListOptions,
+) error {
+	logMessage := fmt.Sprintf("Waiting for all pods in %v namespaces", nsNames)
+	passedOptions := metav1.ListOptions{}
+
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return fmt.Errorf("podList 'apiClient' cannot be empty")
+	}
+
+	if len(options) > 1 {
+		glog.V(100).Infof("'options' parameter must be empty or single-valued")
```

Comment:
shouldn;t the options be option (single) if it only supports a single option

Reply:
I followed the pattern in other fucntions such as ListInAllNamespaces, etc